pymeasure/display/inputs.py

```diff
@@ -41,10 +41,20 @@ def __init__(self, parameter):
         super().__init__()
         self._parameter = None
         self.set_parameter(parameter)
+        self.set_value(parameter)
 
     def set_parameter(self, parameter):
         self._parameter = parameter
 
+    def set_value(self, parameter):
+        if hasattr(self, 'setValue'):
+            try:
+                value = parameter.value  # if parameter is not set yet, this will raise ValueError
+            except ValueError:
+                pass  # parameter is not set yet
+            else:
```

Comment:
OK.
ad 3) that's actually a good reason to do that with an else. I didn't know if that was a relevant scenario, though, that's why I asked.